### DISCUSSION TITLE
Deploy aplikasi Samudra

### DISCUSSION THREAD

**Discussion by alserembani94** [samudradev/samudra#10] [General]
Bila kita boleh membuat penyebaran? Mungkin boleh fikirkan beberapa pilihan:

1. Heroku - boleh juga menggunakan plugin `postgre` dalam Heroku jika perlu
2. Vercel - tapi tak pasti sama ada Vercel sokong poetry atau tidak, boleh cuba juga
3. Netlify - lebih kurang seperti Vercel juga
4. Digital Ocean - tapi tidak percuma lah, jika untuk tujuan percubaan.

**Comment by Thaza-Kun** [comment]
Ada dua tempat yang aku fikirkan boleh dijadikan perumah bagi pelayan kita:
1. [Railway.app](https://railway.app/)
2. [Fly.io](https://fly.io/)

Aku sudah sediakan `Dockerfile` yang aku uji pakai Railway. Fly juga rasanya pakai `Dockerfile`.

Heroku sudah tiada lagi khidmat percuma (https://blog.heroku.com/next-chapter).
Aku sendiri belum cuba pakai Vercel untuk ini. Netlify pun sama belum.
Aku pernah cuba Digital Ocean tapi entah tak faham apa masalahnya jadi aku abaikan je sebab aku dah berjaya pakai Railway.

Untuk bahagian pangkalan data, ada dua tempat aku nampak:
1. [PlanetScale](https://planetscale.com) (MySQL). Aku tertarik dengan dia punya model macam git tapi untuk db. Cuma harganya mahal kalau masuk yang berbayar, tapi khidmat yang percuma dia pun boleh tahan, lebih 10 milion R/W sebulan. Jadi, mungkin boleh digunakan pada tahap awal tak ramai pengguna.
2. [CockroachDB](https://www.cockroachlabs.com/product/) (PostgreSQL). Buat masa ini takde apa yang menarik pasal CockroachDB bagi aku. Harganya pula ikut penggunaan. Khidmat percuma dia pun boleh tahan dengan 250 R/W sebulan. Akan tetapi, aku masih tak berapa faham macam mana nak berinteraksi dengan DBnya. Jadi, aku akan lebih cenderung ke PScale sebab sekali pandang nampak macam arahannya mudah faham walaupun aku belum cuba.

Kalau ada lagi penyedia pangkalan data (DBaaS) lain yang dirasakan boleh digunakan, boleh sahaja cadang.

**Reply by Thaza-Kun** [reply]
Nampaknya ada jumpa beberapa contoh yang lancarkan FastAPI ikut Vercel. 
- https://blog.logrocket.com/deploying-fastapi-applications-to-vercel/
- https://github.com/marcorichetta/fastapi-vercel
- https://betterprogramming.pub/how-to-build-and-deploy-a-fastapi-task-manager-app-on-vercel-c3aa82b8365e

Barangkali lebih baik guna Vercel berbanding yang lain untuk kurangkan masa komunikasi API. Baiklah, aku akan fokus dapatkan cara guna Vercel. Bab Poetry itu, rasanya boleh sahaja. Secara teorinya kita boleh guna pip dengan menghasilkan `requirements.txt` melalui `poetry export`.

Apa-apa pun nanti aku bagi berita lanjut.
